cmd/config.go

```diff
@@ -309,6 +301,85 @@ type CAADistributedResolverConfig struct {
 	Proxies     []string
 }
 
+type TemporalLogDescription struct {
+	URI         string
+	Key         string
+	WindowStart string
+	start       time.Time
+	WindowEnd   string
+	end         time.Time
+}
+
+type TemporalSet struct {
+	Name   string
+	Shards []TemporalLogDescription
+}
+
+// Setup initializes the TemporalSet by parsing the start and end dates
+// and verifying the intervals
+func (ts *TemporalSet) Setup() error {
+	if ts.Name == "" {
+		return errors.New("Name cannot be empty")
+	}
+	var err error
+	for i := range ts.Shards {
+		ts.Shards[i].start, err = time.Parse(time.RFC3339, ts.Shards[i].WindowStart)
+		if err != nil {
+			return err
+		}
+		ts.Shards[i].end, err = time.Parse(time.RFC3339, ts.Shards[i].WindowEnd)
+		if err != nil {
+			return err
+		}
+		if ts.Shards[i].end.Before(ts.Shards[i].start) || ts.Shards[i].end.Equal(ts.Shards[i].start) {
+			return errors.New("WindowStart must be before WindowEnd")
+		}
+	}
+	return nil
+}
+
+func (ts *TemporalSet) pick(exp time.Time) (*TemporalLogDescription, error) {
+	for _, shard := range ts.Shards {
+		if exp.Before(shard.start) {
+			continue
+		}
+		if !exp.Before(shard.end) {
+			continue
+		}
+		return &shard, nil
+	}
+	return nil, fmt.Errorf("no valid shard available for temporal set %q", ts.Name)
```

Comment:
I think this error return should also include the `exp` date since that's the key part of why no valid shard was available in the temporal set.